### PR TITLE
Successful implementation of GMAIL API for automated confirmation email

### PR DESCRIPTION
**Pull request by Sedwards8900**

### CODE REVIEW

File: src/app/routes/api/gmail.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+from googleapiclient.discovery import build
+from apiclient import errors
+from email.mime.text import MIMEText
+import pybase64
+from google.oauth2 import service_account
+from googleapiclient.discovery import build
+
+
+# Email sending confirmation email
+EMAIL_FROM = "admin@piecemealapp2.com"
```
Comment (williamgarland) [inline src/app/routes/api/gmail.py]: This should not be visible in the source code. I'd recommend making it an environment variable.

File: src/app/routes/api/gmail.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+    Args:
+        New user email address provided during signup process
+
+    """
+
+    # Email from user must be retrieved and stored in EMAIL_TO
+    EMAIL_TO = new_user_email
+    EMAIL_SUBJECT = "Welcome to pieceMeal!"
+    EMAIL_CONTENT = "Welcome to pieceMeal! \nThank you for creating a new account with us. \nPlease sign in at https://www.piecemealapp2.com to complete your profile"
```
Comment (williamgarland) [inline src/app/routes/api/gmail.py]: This line is too long. Make sure you run the linter (PyLint) to see if you have any linting errors

File: src/app/routes/api/misc.py [hunk 3]
```diff
@@ -12,6 +12,7 @@
 from Crypto.PublicKey import RSA
 from Crypto.Hash import SHA256
 
+from ...routes.api.gmail import confirmation_email
```
Comment (williamgarland) [inline src/app/routes/api/misc.py]: This import won't work for all Python interpreters.

File: src/app/routes/api/gmail.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+import pybase64
+from google.oauth2 import service_account
+from googleapiclient.discovery import build
+
+
+# Email sending confirmation email
+EMAIL_FROM = "admin@piecemealapp2.com"
+
+
+def confirmation_email(new_user_email):
```
Comment (williamgarland) [inline src/app/routes/api/gmail.py]: I'd recommend renaming this function to something starting with a verb, such as `send_confirmation_email`

**Review by williamgarland — COMMENTED** [review]
The `gmail.py` file should be moved to `/src/api` since it doesn't have any routes in it. This means you'll need to change like 15 in `misc.py` to be `from ...api.gmail import confirmation_email`.

File: src/app/routes/api/misc.py (hunk 3 above)
Comment (Sedwards8900) [reply src/app/routes/api/misc.py]: when I add gmail directly or ...api.email the file gives me an error saying unable to import file

**Pull request merged**